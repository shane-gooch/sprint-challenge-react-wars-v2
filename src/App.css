@import url("https://fonts.googleapis.com/css?family=Economica|Roboto+Condensed");

body {
  text-align: center;
  font-family: "Roboto Condensed", sans-serif;
}

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  margin-top: 20px;
  margin-bottom: 20px;
  max-width: 950px;
  text-align: center;
}

.StarWarsCharContainer {
  width: 75%;
}
