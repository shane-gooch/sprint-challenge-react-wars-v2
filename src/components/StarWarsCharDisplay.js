import React, { useState } from "react";
import FilmList from "./FilmList";

import "semantic-ui-css/semantic.min.css";
import { Segment, Button } from "semantic-ui-react";

export default function StarWarsCharDisplay(props) {
  const [showFilms, setShowFilms] = useState(false);

  const showFilmsHandler = () => {
    setShowFilms(!showFilms);
  };

  return (
    <div className="StarWarsCharContainer">
      {props.starWarsChars.map(char => {
        return (
          <Segment.Group>
            <div key={char.name}>
              <Segment>
                <h3>{char.name}</h3>
              </Segment>
              <Segment>
                <p>{char.birth_year}</p>
              </Segment>
              <Button onClick={showFilmsHandler} color="blue">
                Show Films
              </Button>
              {showFilms ? <FilmList films={char.films} /> : null}
            </div>
          </Segment.Group>
        );
      })}
    </div>
  );
}
