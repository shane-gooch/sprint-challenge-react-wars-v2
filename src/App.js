import React, { useState, useEffect } from "react";
import axios from "axios";

import "./App.css";
import "semantic-ui-css/semantic.min.css";
import { Button } from "semantic-ui-react";

import StarWarsCharDisplay from "./components/StarWarsCharDisplay";

function App() {
  const [starWarsChars, setstarWarsChars] = useState([]);
  const [nextPageState, setNextPageState] = useState({});
  const [prevPageState, setPrevPageState] = useState({});
  const [nextPage, setNextPage] = useState(false);
  const [prevPage, setPrevPage] = useState(false);
  useEffect(() => {
    const getStarWarsChars = () => {
      axios
        .get("https://swapi.co/api/people/")
        .then(res => {
          setstarWarsChars(res.data.results);
          setNextPageState(res.data.next);
        })
        .catch(err => console.log(err));
    };
    getStarWarsChars();
  }, []);
  useEffect(() => {
    const getNextPage = () => {
      axios
        .get(nextPageState)
        .then(res => {
          setstarWarsChars(res.data.results);
          setNextPageState(res.data.next);
          setPrevPageState(res.data.previous);
        })
        .catch(err => console.log(err));
    };
    getNextPage();
  }, [nextPage]);
  useEffect(() => {
    const getPrevPage = () => {
      axios
        .get(prevPageState)
        .then(res => {
          setstarWarsChars(res.data.results);
          setNextPageState(res.data.next);
          setPrevPageState(res.data.previous);
        })
        .catch(err => console.log(err));
    };
    getPrevPage();
  }, [prevPage]);

  const nextPageHandler = () => {
    setNextPage(!nextPage);
  };
  const prevPageHandler = () => {
    setPrevPage(!prevPage);
  };
  return (
    <div className="App">
      <h1>React Wars</h1>
      <Button color="blue" onClick={prevPageHandler}>
        Previous
      </Button>
      <Button color="blue" onClick={nextPageHandler}>
        Next
      </Button>
      <StarWarsCharDisplay starWarsChars={starWarsChars} />
    </div>
  );
}

export default App;
