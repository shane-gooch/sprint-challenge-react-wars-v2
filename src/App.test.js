import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import StarWarsCharDisplay from "../src/components/StarWarsCharDisplay";

import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("<App />", () => {
  it("Should Match SnapShot", () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("<StarWarsCharDisplay />", () => {
  it("Should Match SnapShot", () => {
    const tree = renderer.create(<StarWarsCharDisplay />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
