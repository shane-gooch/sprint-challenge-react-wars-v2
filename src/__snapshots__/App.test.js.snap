// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should Match SnapShot 1`] = `
<div
  className="App"
>
  <div
    className="StarWarsCharContainer"
  >
    <h1>
      React Wars
    </h1>
  </div>
</div>
`;
